contrib/babelfishpg_tds/src/include/tds_request.h

```diff
@@ -269,6 +288,104 @@ typedef TDSRequestData *TDSRequest;
 #define TVP_COLUMN_ORDERING_TOKEN		0x11
 #define TVP_END_TOKEN				0x00
 
+/* Find namespace oid of a procedure based on proc name. */
+static inline Oid
+get_proc_namespace_oid(char **proc_name, char *curr_db)
+{
+	char *logical_sch_name;
+	char *physical_sch_name;
+	char *db_name;
+	char *schema_name;
+	char	  **splited_object_name;
+	Oid obj_schema_oid = InvalidOid;
+
+	/* resolve the three part name */
+	splited_object_name = pltsql_plugin_handler_ptr->split_object_name(*proc_name);
+	db_name = splited_object_name[1];
+	schema_name = splited_object_name[2];
+	*proc_name = splited_object_name[3];
+
+	if (!strcmp(db_name, ""))
+		db_name = curr_db;
+
+	if (!strcmp(schema_name, ""))
+	{
+		/* Find the default schema for current user and get physical schema name */
+		const char *user = pltsql_plugin_handler_ptr->pltsql_get_user_for_database(db_name);
+		schema_name = pltsql_plugin_handler_ptr->get_authid_user_ext_schema_name((const char *) db_name, user);
+	}
+
+	logical_sch_name = downcase_truncate_identifier(schema_name,strlen(schema_name), true);
+	/* Get physical schema name from logical schema name */
+	physical_sch_name = pltsql_plugin_handler_ptr->get_physical_schema_name(db_name, logical_sch_name);
+	/* Get namespace oid from physical schema name */
+	obj_schema_oid = get_namespace_oid(physical_sch_name, false);
+
+	return obj_schema_oid;
+
+}
+
+static inline Oid
+tds_get_proargtypes_oid(char *proname, Oid pronamespace, Oid user_id, char *targeted_arg_name)
+{
+	HeapTuple    tuple;
+	CatCList   *catlist;
+	Oid matched_type = InvalidOid;
+	char *hash;
+	char new_target_name[NAMEDATALEN];
+
+	if (strlen(targeted_arg_name) >= NAMEDATALEN)
+	{
+		/* Truncate the targeted_arg_name to 31 characters */
+		strncpy(new_target_name, targeted_arg_name, NAMEDATALEN - MD5_HASH_LEN - 1);
+		new_target_name[NAMEDATALEN - MD5_HASH_LEN - 1] = '\0';
+		/* Generate the hash */
+		hash = pltsql_plugin_handler_ptr->construct_unique_hash(targeted_arg_name);
+		/* Append the hash to the truncated name */
+		strcat(new_target_name, hash);
+		targeted_arg_name = new_target_name;
+	}
+
+	/*
+	 * Search pg_proc for the procedure by name in the specified namespace and
+	 * return the argument type, if the targeted argument name matches
+	 */
+	catlist = SearchSysCacheList1(PROCNAMEARGSNSP, CStringGetDatum(proname));
+
+	for (int i = 0; i < catlist->n_members; i++)
+	{
+		Form_pg_proc procform;
+
+		tuple = &catlist->members[i]->tuple;
+		procform = (Form_pg_proc) GETSTRUCT(tuple);
+
+		if (procform->pronamespace == pronamespace &&
+			object_aclcheck(ProcedureRelationId, procform->oid, user_id, ACL_EXECUTE) == ACLCHECK_OK)
+		{
+			char **proargnames = pltsql_plugin_handler_ptr->fetch_func_input_arg_names(tuple);
+			Oid *proargtypes = procform->proargtypes.values;
+
+			for (int j = 0; j < procform->pronargs; j++)
+			{
+				if (strcmp(proargnames[j], targeted_arg_name) == 0)
```

Comment:
Lets add tests with same name of args in the procedure, if its possible

Reply:
A procedure can't have two arguments with same name. 
Babelfish 
```
1> CREATE PROCEDURE proc21  @t2 t1 READONLY, @t2 INT AS BEGIN SET NOCOUNT ON; select 1 end
2> go
Msg 33557097, Level 16, State 1, Server BABELFISH, Line 1
parameter name "@t2" used more than once
```
TSQL
```
1> CREATE PROCEDURE proc21  @t2 t1 READONLY, @t2 INT AS BEGIN SET NOCOUNT ON; select 1 end
2> go
Msg 134, Level 15, State 1, Server EC2AMAZ-EULFEOJ, Procedure proc21, Line 1
The variable name '@t2' has already been declared. Variable names must be unique within a query batch or stored procedure.
```